src/main/java/io/cdap/delta/bigquery/BigQueryUtils.java

```diff
@@ -198,20 +186,17 @@ public static String normalizeTableName(String name) {
    * @return the normalized name
    */
   public static String normalizeFieldName(String name) {
-    return normalize(name, FIELD_NAME_MAX_LENGTH, false, false);
+    return normalize(name, FIELD_NAME_MAX_LENGTH, false);
   }
 
-  private static String normalize(String name, int maxLength, boolean canStartWithNumber, boolean useExtendedCharset) {
+  private static String normalize(String name, int maxLength, boolean canStartWithNumber) {
     if (name == null || name.isEmpty()) {
       return name;
     }
 
     // replace invalid chars with underscores if there are any
-    if (useExtendedCharset && !VALID_TABLE_NAME_REGEX.matcher(name).matches()) {
-      name = INVALID_TABLE_NAME_REGEX.matcher(name).replaceAll("_");
-    }
-    if (!useExtendedCharset && !VALID_DATASET_NAME_REGEX.matcher(name).matches()) {
-      name = INVALID_DATASET_NAME_REGEX.matcher(name).replaceAll("_");
+    if (!name.matches(VALID_NAME_REGEX)) {
```

Comment:
Using a `Pattern` is more efficient here, as otherwise `String.matches()` internally will create and compile a new `Pattern` object internally every time, please refer to https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html#matches-java.lang.String-java.lang.CharSequence-